index.html

```diff
@@ -0,0 +1,75 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Number Facts</title>
+    <link
+      href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/css/bootstrap.min.css"
+      rel="stylesheet"
+      integrity="sha384-T3c6CoIi6uLrA9TneNEoa7RxnatzjcDSCmG1MXxSR1GAsXEV/Dwwykc2MPK8M2HN"
+      crossorigin="anonymous"
+    />
+    <style>
+      #fact {
+        display: none;
+      }
+    </style>
+  </head>
+  <body class="bg-dark">
+    <div class="container">
+      <div class="row">
+        <div class="col-md-6 mx-auto">
+          <div class="card bg-primary text-white mt-5 p-4">
+            <h1>Number Facts</h1>
+            <p>Enter a number and get a random fact</p>
+            <input
+              type="number"
+              class="form-control form-control-lg"
+              id="numberInput"
+              placeholder="Enter a number..."
+            />
+            <div id="fact" class="card-body">
+              <h4 class="card-title">Number Fact</h4>
+              <p id="factText" class="card-text"></p>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+    <script>
+      let factCont = document.getElementById("fact");
+      let factText = document.getElementById("factText");
+
+      let numInput = document.getElementById("numberInput");
+      numInput.addEventListener("input", getFactFetch);
+
+      function getFactFetch() {
+        let number = numInput.value;
+        fetch("http://numbersapi.com/" + number)
+          .then((response) => response.text())
+          .then((data) => {
+            if (number != "") {
+              factCont.style.display = "block";
+              factText.innerText = data;
+            } else {
+              factCont.style.display = "none";
+            }
+          })
+          .catch((err) => console.log(err));
+      }
+
+      async function getFactFetch2() {
+        let number = numInput.value;
+        let result = await fetch("http://numbersapi.com/" + number);
+        let resultFact = await result.text();
+        factText.innerText = resultFact;
+        factCont.style.display = "block";
+
+        if (number === "") {
+          factCont.style.display = "none";
+        }
```

Comment:
I like both of your implementations of `getFactFetch` 👍 .
A little comment regarding `factCont.style.display`. For such cases when we want to assign one or another value to some variable depending on some condition there is a commonly used ternary operator in JS (you can read more about it here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator)
Long story short, you can simplify this part of code the following way:
```
factCont.style.display = number === "" ? "none" : "block";
```